app/src/main/java/org/supla/android/lib/SuplaClient.java

```diff
@@ -764,7 +771,9 @@ private void onConnError(SuplaConnError connError) {
         msg.setConnError(connError);
         sendMessage(msg);
 
-        if (connError.Code == SuplaConst.SUPLA_RESULTCODE_HOSTNOTFOUND) {
+        String host = this.getHost();
+
+        if (connError.Code == SuplaConst.SUPLA_RESULTCODE_HOSTNOTFOUND || !host.contains(".supla.org")) {
```

Comment:
It is not a good idea to cancel a thread for all version-related bugs. There are errors that require reconnection.

Reply:
I get it, but it's also the core of the problem. As a user I got to experience a few times getting stuck indefinitely on the "Connecting..." screen. Potentially we could add an error counter and once a limit of errors is reached (e.g. 10 errors per connection attempt), we show the error screen to the user. In that case we could also probably remove my host name check. What do you think?

Reply:
This PR solves one problem and creates another. What was the real reason the application was unable to connect to your server?

Reply:
Closed port 2015. It threw error: Can't connect to host